templates/toil-instance.yaml

```diff
@@ -246,8 +247,76 @@ Resources:
               group: "root"
             /opt/toil/bin/reload-toil-cluster.sh:
               content: |
-                /opt/toil/bin/remove-toil-cluster.sh
-                /opt/toil/bin/create-toil-cluster.sh
+                /bin/bash /opt/toil/bin/remove-toil-cluster.sh
+                /bin/bash /opt/toil/bin/create-toil-cluster.sh
+              mode: "00644"
+              owner: "root"
+              group: "root"
+            /opt/toil/bin/add-security-group.py:
+              content: !Sub
+                - |
+                  #! /usr/bin/env python
+
+                  import boto3
+
+                  region = 'us-east-1'
+                  client = boto3.client('ec2', region_name=region)
+
+                  print('Ensure vpc security group is associated with leader node.')
+
+                  print('\nSending describe instances request')
+                  describe_instances_response = client.describe_instances(
+                    Filters=[
+                      {
+                          'Name': 'tag:Name',
+                          'Values': [
+                              "${ClusterName}",
+                          ]
+                      }
+                    ])
+
+                  print('Describe instances response:'')
+                  print(describe_instances_response)
+
+                  instance_ids = [
```

Comment:
are you querying all instances?  Not sure why you need to do that.  I thought you only cared about a single leader node?  or could there be multiple leaders?

Reply:
This is the only way I know of to get at the leader. I don't have the instance id. So I have to look at what is made and find it by name.

Reply:
could you not filter by `ToilNodeType`?
```
>> aws ec2 describe-instances --output table --instance-id i-0e6220fc3812e20dd|grep Tags -C40

|||+-----------------------------------------------------+---------------------------------------------------------------------------+|||
||||                                                              Tags                                                               ||||
|||+------------------------------+--------------------------------------------------------------------------------------------------+|||
||||              Key             |                                              Value                                               ||||
|||+------------------------------+--------------------------------------------------------------------------------------------------+|||
||||  Owner                       |  toil                                                                                            ||||
||||  Department                  |  SysBio                                                                                          ||||
||||  SubProject                  |  amp-ad-workflows                                                                                ||||
||||  Project                     |  amp-ad                                                                                          ||||
||||  ToilNodeType                |  leader                                                                                          ||||
||||  OwnerEmail                  |  tess.thyer@sagebase.org                                                                         ||||
||||  parkmycloud                 |  yes                                                                                             ||||
||||  Name                        |  rna-seq-reprocessing-scicomp-toil-cluster-v001                                                  ||||
|||+------------------------------+--------------------------------------------------------------------------------------------------+|||
```

Reply:
Good find -- I will use name and ToilNodeType.